src/main/kotlin/gg/essential/elementa/WindowScreen.kt

```diff
@@ -65,8 +67,22 @@ abstract class WindowScreen @JvmOverloads constructor(
     override fun onMouseClicked(mouseX: Double, mouseY: Double, mouseButton: Int) {
         super.onMouseClicked(mouseX, mouseY, mouseButton)
 
+        // Restore decimal value to mouse locations if not present.
+        // See [ElementaVersion.V2] for more info
+        val (adjustedMouseX, adjustedMouseY) = if (
+            version >= ElementaVersion.V2 &&
+            (mouseX == floor(mouseX) && mouseY == floor(mouseY))
+        ) {
+            val x = UMouse.Scaled.x
+            val y = UMouse.Scaled.y
+
+            mouseX + (x - floor(x)) to mouseY + (y - floor(y))
+        } else {
+            mouseX to mouseY
+        }
```

Comment:
The formatting of that `if` condition feels really odd. If the initial line is already too long, then the standard way to format that would be to put it into the next line (and ktfmt would agree):
```suggestion
    val (adjustedMouseX, adjustedMouseY) =
        if (version >= ElementaVersion.v2 && (mouseX == floor(mouseX) && mouseY == floor(mouseY))) {
          val x = UMouse.Scaled.x
          val y = UMouse.Scaled.y

          mouseX + (x - floor(x)) to mouseY + (y - floor(y))
        } else {
          mouseX to mouseY
        }
```

Also, you should use the `v2` (lowercase `v`) for comparisons within Elementa, otherwise we'll have a bunch of deprecated code everywhere once `V2` becomes deprecated.